src/util/train.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include "gta/enums.hpp"
+#include "services/players/player_service.hpp"
+#include "script_global.hpp"
+#include "pointers.hpp"
+#include "natives.hpp"
+
+namespace big::train
+{
+	inline auto get_all_vehicles = []() -> std::vector<Vehicle>
+	{
+		std::vector<Vehicle> result;
+		Vehicle vehs[300 * 2 + 2];
+		vehs[0] = 300;
+		int nums = VEHICLE::GET_ALL_VEHICLES(vehs);
+		for (int i = 1; result.size() < nums && i < sizeof(vehs) / sizeof(vehs[0]); i++)
+		{
+			if (ENTITY::DOES_ENTITY_EXIST(vehs[i]) && !ENTITY::IS_ENTITY_DEAD(vehs[i], false))
+			{
+				result.push_back(vehs[i]);
+			}
+		}
+		return result;
+	};
```

Comment:
Why is index[0] set to `300`?
Why does the iterator start at 1?

`std::vector::size()` is a method returns the amount of entries in a vector.

Try taking a look at how the replay interface works, you can easily iterate of all the entities that exist on your local machine and push them onto a vector.

Reply:
that function was written by one the guys above

Reply:
I'll take a look at replacing it with the replay interface.

Reply:
better?